### ISSUE TITLE
Promscale should not terminate under non-critical conditions

### ISSUE THREAD

**Issue by mjf** [timescale/promscale#592]
*Promscale* terminates with error under conditions when one would expect just some log entries. For example when it could not connect to a database (this may be because of many reasons and have simply nothing to do with *Promscale* as such, *i.e.* somebody may have deployed new *Promscale* instance and have not yet configured *PostgreSQL* role for *Promscale* for it, or even hasn't created the database yet, or there is `pg_hba.conf` entry missing etc.). I agree that software in general should terminate if there is a critical configuration error or serious runtime error but I do not agree it should terminate if the other side is just rejecting connections for any reason etc. So what I suggest is to make database connection errors **non-critical** so that normal connection retries apply. Optionally some option may be provided to influence the behaviour for backward compatibility.

**Comment by Harkishen-Singh** [comment]
I agree we shouldn't terminate on all conditions. It would be better to categorize some as recoverable and some as non-recoverable and then terminal only on non-recoverable and the rest can be a retry.

**Comment by VineethReddy02** [comment]
closing this issue, as we haven’t received multiple requests from users for this feature and it's quite stale! :)

If you still have a need for this, Please open a new issue.

Thanks!
